src/components/grader/submissions/CodeEditor.js

```diff
@@ -0,0 +1,159 @@
+// import React, { useState, useRef } from "react";
+import React, { useState, useEffect } from 'react';
+import MultiRef from 'react-multi-ref';
+import { Tabs, Tab } from 'carbon-components-react';
+import AceEditor from "react-ace";
+import { ModalWrapper } from 'carbon-components-react';
+import "ace-builds/src-noconflict/mode-python";
+import "ace-builds/src-noconflict/mode-sql";
+import "ace-builds/src-noconflict/mode-c_cpp";
+import "ace-builds/src-noconflict/mode-java";
+import "ace-builds/src-noconflict/mode-javascript";
+import "ace-builds/src-noconflict/mode-json";
+
+import "ace-builds/src-noconflict/theme-dracula";
+import 'ace-builds/src-min-noconflict/ext-searchbox';
+import apiClient from "../../../api-client";
+
+
+function CodeEditor(props) {
+    const [itemRefs] = useState(() => new MultiRef());
+    const [tabRefs] = useState(() => new MultiRef());
+    const [preVal, setPreVal] = useState("");
+    // eslint-disable-next-line
+    const [theLabels, setTheLabels] = useState([])
+    const [currentTab, setCurrentTab] = useState(0)
+    const [currentTabLabel, setCurrentTabLabel] = useState(props.subList[0])
+    // eslint-disable-next-line
+    const [submissionFiles, setSubmissionFiles] = useState({})
+
+
+    useEffect(() => {
+        if (isEmpty()) {
+            // fetch file here 
+            apiClient.getFile(props.course, props.lab, props.submissionId, currentTabLabel)
+                .then(res => {
```

Comment:
please consider using `async` / `await` syntax

Reply:
- [x] added to my todo list